gateway-server/src/main/java/org/apache/knox/gateway/services/factory/RemoteRegistryClientServiceFactory.java

```diff
@@ -26,20 +26,26 @@
 import org.apache.knox.gateway.services.ServiceLifecycleException;
 import org.apache.knox.gateway.services.ServiceType;
 import org.apache.knox.gateway.services.config.client.RemoteConfigurationRegistryClientService;
+import org.apache.knox.gateway.services.security.AliasService;
 
 // There are two implementations of 'RemoteConfigurationRegistryClientService':
 // - LocalFileSystemRemoteConfigurationRegistryClientService
 // - CuratorClientService
 // However, the first one - local - is for test-only purposes
 public class RemoteRegistryClientServiceFactory extends AbstractServiceFactory {
 
+  private final AliasServiceFactory aliasServiceFactory = new AliasServiceFactory();
+
   @Override
   public Service create(GatewayServices gatewayServices, ServiceType serviceType, GatewayConfig gatewayConfig, Map<String, String> options, String implementation)
```

Comment:
Can this implementation not be overridden?

Reply:
I'm not sure if I understood your comment here. Could you please elaborate?

Reply:
This is related to my most recent comment, I guess. Since there isn't another known implementation, it currently cannot be overridden. So, the heart of the matter is the ability to plug-in an arbitrary gateway service implementation.

Reply:
Fixed.